client/src/components/GoogleMap.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+  AdvancedMarker,
+  AdvancedMarkerAnchorPoint,
+  AdvancedMarkerRef,
+  InfoWindow,
+  Map,
+  useMapsLibrary,
+  useMap,
+  Pin,
+} from '@vis.gl/react-google-maps';
+import { SetStateAction, useCallback, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import './GoogleMap.css';
+
+type PlaceResult = {
+  place_id?: string;
+  name?: string;
+  address?: string;
+  vicinity?: string;
+  rating?: number;
+};
+
+export function GoogleMap() {
+  const { therapyType } = useParams();
+  const { marker, refCallback } = useAdvancedMarkerRef();
+  const [isOpen, setIsOpen] = useState(false);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<unknown>();
+  const [placesService, setPlacesService] = useState<
+    google.maps.places.PlaceResult[]
+  >([]);
+
+  const [selectedPlace, setSelectedPlace] = useState<PlaceResult>();
+
+  const map = useMap();
+  const placesLibrary = useMapsLibrary('places');
+
+  useEffect(() => {
+    if (!placesLibrary || !map || !therapyType) {
+      return;
+    }
+
+    setIsLoading(true);
+    setError(null);
+
+    try {
+      const svc = new placesLibrary.PlacesService(map);
+
+      const request: google.maps.places.PlaceSearchRequest = {
+        location: map.getCenter(),
+        radius: 10000,
+        type: 'psychotherapy',
+        keyword: `${therapyType}`,
+      };
+
+      svc.nearbySearch(request, (therapyType, status) => {
+        if (status === placesLibrary.PlacesServiceStatus.OK && therapyType) {
+          setPlacesService(therapyType);
+        } else {
+          setError(
+            `No results found for ${therapyType} therapists in this area. `
+          );
+        }
+
+        setIsLoading(false);
+      });
+    } catch (err) {
+      setIsLoading(false);
+      setError(
+        `An unexpected error occurred: ${
+          err instanceof Error ? err.message : String(err)
+        }`
+      );
+    }
+  }, [placesLibrary, map, therapyType]);
+
+  function useAdvancedMarkerRef() {
+    const [marker, setMarker] = useState<AdvancedMarkerRef | null>(null);
+    const refCallback = useCallback((m: AdvancedMarkerRef | null) => {
+      setMarker(m);
+    }, []);
+    return { marker, refCallback };
+  }
+
+  const handleMarkerClick = (
```

Comment:
You should be more consistent with how you define functions inside your components. Either always use `function` syntax or always use arrow function syntax.